### DISCUSSION TITLE
Definition of and comments for 'is a measurement of

### DISCUSSION THREAD

**Discussion by alanruttenberg** [CommonCoreOntology/CommonCoreOntologies#893] [Ontology Terms and Class Schema]
I think this is confusing. First, a measurement would commonly be understood as being the output of making a measurement. In that sense, the comment makes sense. If you are making a measurement of the size of a tire, it is the measurement made of that specific tire. However, consider the specification of that tire. The definition would seem to hold between that specification and any tire of that model, in which case it would not be functional.

I think that the definition should be clarified to make clear that the information entity is the output of a measurement process, since we want to represent that, and that it is the measurement of that particular entity. I think there should be further alignment with OBI. Specifically they have a measurement process called 'assay' - obviously in this community it should have a different label. 'Measurement Artifact Function' would have this as a realization.
OBI also has a more general relation 'has specified value' that is closer to the current sense of 'is measurement of' in that it is valid for measurements, specifications, and predictions - all of which have the same *shape* in that they are e.g. often pairs of value and unit.
CCO lacks an information entity that is specifically a measurement. OBO uses the class 'value specification'.

Summary: 
1. Add 'measurement process', realization of 'measurement artifact function'
2. Add 'value specification' (ICE)
3. Add 'has specified value' (and inverse) (superproperty of 'measured by', subproperty of 'is described by'
4. Clarify that 'is measurement of' is output of a 'measurement process'

Definition: x is_a_measurement_of y iff x is an instance of Information Content Entity and y is an instance of Entity, such that x describes some attribute of y relative to some scale or classification scheme.

Comment: This object property, as well as all of its children are typified as functional properties. This means that for instances x, y, and z if x is a measurement of y and x is a measurement of z, then y = z.

**Comment by cameronmore** [comment]
Isn't the measurement of a specification still functional since it's a measurement of the Tire Artifact Model?

So, suppose that `Goodyear Tire Artifact Model` `has part` `Tire Width` and there's a measure of that `Tire Width` that has the value `215` which uses the `Millimeter Measurement Unit`. That measurement is a measurement of some portion of the Goodyear Tire Artifact Model, not a measurement of all tires produced according to that model. That same measurement (of 215) `is_measurement_of` the Artifact Model as well as `prescribes` all instances of Tire which are the output of an Act of Manufacturing which is done according to that Goodyear Tire Artifact Model (which could be captured with an equivalence class).

**Comment by alanruttenberg** [comment]
There are no measurements of "models" as a "model" isn't a singular physical thing. There are various interpretations of what a model is. It could be considered the class of tires, or it could be considered the design, an information artifact, with the class of tires those tires that the design prescribes.  The numbers associated with the design might be called measurements, but they are really specifications. At the time the specifications are written there are no size qualities around that the measurement could be about.

**Comment by alanruttenberg** [comment]
Also, pay attention to the definition: "x is_a_measurement_of y iff x is an instance of Information Content Entity and y is an instance of Entity, such that x describes some attribute of y relative to some scale or classification scheme."
If a specification for a model is properly executed - the artifacts conform to the model, then it is reasonable to say x is the specification and y is a tire instance, and the definition holds. The specified size describes the actual size of the tires, otherwise the advertisement I pointed to would not be something that could be the basis for getting the tire you need.

**Comment by cameronmore** [comment]
That is a good point about the model , and my interpretation of ‘attributes of an entity’ started to wade into modal territory and a little bit outside BFO’s scope of attribute.

Are all measurements the output of an Act of Measuring? If so, what would be the act of Measuring in the case where you create a measurement for a tire specification, and only afterwards create the instances of the tires--it seems like that would be a 'retroactive measurement'. 

Another way of retaining the functional characteristic of measurements in this case is to suggest that the measurement is really about an aggregative tires— thus still a single entity which might have multiple other entities as members.

One thing I’m still thinking through is the difference between measuring and prescribing. It seems like a specification measurement is a measurement that prescribes, and prescribes is not functional. Which leads me to think that what we’ve been calling a measurement is really not a measurement in the ontological sense, but just a prescriptive information content entity whose content ‘looks’ like a measurement (215 millimeters).

**Comment by alanruttenberg** [comment]
All measurements in the usual sense and in the sense I intend are outputs of acts of measurement as defined in CCO. One entity, some process that produces information about it.
>If so, what would be the act of Measuring in the case where you create a measurement for a tire specification, and only afterwards create the instances of the tires--it seems like that would be a 'retroactive measurement'.

It would not be an act of measurement. You did not create a measurement for the tire specification, you created a specification of e.g. a length quality.
>Another way of retaining the functional characteristic of measurements in this case is to suggest that the measurement is really about an aggregative tires— thus still a single entity which might have multiple other entities as members

You are trying too hard to stretch the meaning to fit. An act of measurement is "A Planned Act that involves determining the extent, dimensions, quanity, or quality of an Entity relative to some standard." If you were measuring an aggregate of tires you should visualize it as putting a pile of tires on a scale. Or you can imagine it as taking individual measurements of each tire and then doing a computation of what the average is. But this is not what the tire specification is.
>One thing I’m still thinking through is the difference between measuring and prescribing.

Good this is what you should be thinking about.
> It seems like a specification measurement 

You are mixing words, rendering each less precise. There is a) specification and b) measurement. These are two different sorts of things. 
>is a measurement that prescribes, and prescribes is not functional. 

Is an *directive information entity* that prescribes and the relation between that information entity and the things it prescribes is not functional.
 
>Which leads me to think that what we’ve been calling a measurement is really not a measurement in the ontological sense, but just a prescriptive information content entity whose content ‘looks’ like a measurement (215 millimeters).

Who has been calling that a measurement? The point of my comment is that there are two separate things and they shouldn't be called by one name. The motivation for value specification is that it would encompass all the various 'measurement shaped' things, on subclass of which are are actual measurements.

Other sorts of things to add to the set prescription (specification) and measurement are predictions and settings, which also can have the same shape but are different sorts of entities. 

For instance, in astronomy you can predict the distance to a star by means of a number of different calculations based on models. Or you can predict some quality of a molecule (say it's dipole moment) based on a hypothesized chemical structure. Neither of those are measurements in the same way putting something on a scale is. In the case of the hypothesized chemical structure there may not even be any instances of it.

For settings the prototypical case is a potentiometer where a certain amount of rotation correlates to the resistance it offers. As part of some plan we would prescribe a setting of a potentiometer (set it for 2.4 ohms)

**Comment by cameronmore** [comment]
> The point of my comment is that there are two separate things and they shouldn't be called by one name. The motivation for value specification is that it would encompass all the various 'measurement shaped' things, on subclass of which are are actual measurements.

I agree, and I was using the confusing language to try to get at the idea that perhaps the same thing is doing two things in different contexts, but this seems wrong. Rather, there are specifications and there are measurements, not one thing. But why would specification be a superclass of measurement if they are distinct things then?

I would be more inclined to add 'Value Specification' to the Directive ICE hierarchy such that _describing_ and _prescribing_ are kept totally separate.

I do acknowledge the problem this would create by having two places in the hierarchy where 'measurement shaped' things are kept, and appreciate the desire to pull these things together.

**Comment by alanruttenberg** [comment]
> But why would specification be a superclass of measurement if they are distinct things then?
It wouldn't. The 'specification' in value specification is not the same sense as 'specification' in directive information content entities, which has a meaning more like 'prescription' or 'specification value'. That would be a subclass of the proposed value specification.

**Comment by cameronmore** [comment]
In the examples above, of the tire specification, the astronomical prediction, and the potentiometer, I'm not sure what value specification adds.

The tire specification is a part of an Artifact Model, a Directive ICE which prescribes the manufacturing of a set of instances of Tire. The astronomical prediction is a modal measurement of some location of astronomical bodies (the Predictive ICE annotations recommend using the modal relation ontology). The setting of the potentiometer is an input to a mechanical process which prescribes how an artifact should act to produce a desired result. In none of these cases are actual material entities being 'measured' in the sense that they are placed on a scale. 

I apologize that I might just be misunderstanding, but what sense of specification does the intended 'value specification' cover if it's distinctly not a specification in the directive ice sense?

Looking at the subclasses of value specification in OBI, it seems as though the thing that a value specification describes is other measurement schemes and classifications, but such a case can already be handled by ordinal measurements. The editor's note for value specification gives the example that "A value specification of '20g' for a measurement data item of the mass of a particular mouse 'is about' the mass of that mouse" but I'm not sure why that wouldn't be a data item?

**Comment by alanruttenberg** [comment]
In all case the thing has a numerical value and unit. Value specification captures that generalization.

**Comment by cameronmore** [comment]
I understand now, thank you. I agree that adding a measurement process would be a good idea.

**Comment by alanruttenberg** [comment]
I had initially missed Act of Measuring that you mentioned above. Maybe that's adequate for measurement process. OBI's assay seems more general on its face, but maybe the CCO definition is adequate.

**Comment by mark-jensen** [comment]
@alanruttenberg I fully agree that the comment re it being a functional property should be removed.

> However, consider the specification of that tire. The definition would seem to hold between that specification and any tire of that model, in which case it would not be functional.

The relation between the spec and a particular tire is directive, not descriptive. As you point out above, there is a difference between a specification and a measurement.
TIRE_WIDTH_SPEC prescribes WIDTH_QUALITY_1234
ICE_1234 is_a_measurment_of WIDTH_QUALITY_1234
WIDTH_QUALITY_1234 inheres_in TIRE_1234

> information entity is the output of a measurement process, since we want to represent that, and that it is the measurement of that particular entity.

I am ok with this. I can’t think of a case when a measurement ICE is not produced by a process of measuring. 

> Add 'measurement process', realization of 'measurement artifact function'

CCO has nominal and ordinal measurement ICEs, e.g, categorizing a threat as weak, strong, critical. The process of producing those kinds of measurements do not realize an artifact function.

We for sure want OBI/IAO and CCO in sync. But, I am not clear how ‘value specification’ and the associated property are used. Can you help me there? What ref. docs. can I look over? What concerns me is that it doesn’t clearly distinguish a prediction from a specification from an actual measurement.

> CCO lacks an information entity that is specifically a measurement.

What do mean by ‘specifically a measurement’?

**Comment by swartik** [comment]
@mark-jensen, you wrote:

    I can’t think of a case when a measurement ICE is not produced by a process of measuring.

What if you:

1. Produce, through an Act of Measuring, a measurement ICE whose `Measurement Unit` is `Kelvin Measurement Unit`;
2. Produce another measurement ICE whose `Measurement Unit` is `Degree Celsius Measurement Unit` by subtracting 273.15 from the value of the first measurement.

Both measure the Temperature of the same independent continuant, but the second is produced through a Planned Act that isn't an Act of Measuring. Or is that too strict an interpretation? I'd like to think not. If someone does the calculation wrong, the ontology still records the original measurement, so you can determine where the error occurred.

**Comment by alanruttenberg** [comment]
@mark-jensen by pointing out the difference between directive and descriptive you highlight the need. value specification sits *below* ICE and *not* in on of the the 4 partitions, in part for this reason. There are other cases we've discussed that don't fit under one of those classes, or fit under both. I tend not to classify under those classes and instead use the relation, because I'm pretty sure that there will be issues of asserted polyhierarchy otherwise.
An instance of value specification might *also* be an instance of one of those classes. 

The alternative would be having a variant of value specification in each of those branches. A constraint like asserting length value specifications can only use a length units shouldn't have to be repeated in the ontology.

Value specification examples:
- length of a chair, a quality, as measured with a tape measure
- The coefficients of a polynomial that characterizes the radial distortion of lens, a quality.
- The proposed length of an artifact being designed, part of plan
- The voltage that needs to be set for a particular experiment, by turning a labeled knob
- The tire size specification above
- A speed limit
- The temperature at which water boils
- The predicted wind speed tomorrow

By particular measurement I mean things like using a device to take measurement the chair in my kitchen, the air speed of a specific aircraft at a certain time at indicated by its wind speed sensor, my recording of the color of the carpet in a house I visited. As opposed to specifications, averages, nominal values, design parameters - everything in my list above but the first example and sometimes the second.

Regarding your temperature example, note that many measurements involve calculations internal to the device making a measurement. Therefore there's no hard boundary between raw and calculated values. In the case you give the transformation is just a unit conversion. Depending on the circumstances  it might be a measurement value, e.g. I have to record heights in meters but I only have a yardstick so I'm calculating metric before I record, or a computed value, the output of a calculation process, another class I think CCO needs. OBI's is called data transformation. Critically, for the purposes of this discussion, whatever the entity is that was created when you added the number, both are value specifications.

**Comment by neilotte** [comment]
@cameronmore Would you mind taking this one on for 2.0?

**Comment by neilotte** [comment]
@cameronmore @mark-jensen  Pushing now to 2.1.
